puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/test_foreman.groovy

```diff
@@ -0,0 +1,199 @@
+pipeline {
+    agent { label 'fast' }
+
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        ansiColor('xterm')
+    }
+
+    stages {
+        stage('Test Matrix') {
+            parallel {
+                stage('ruby-2.5-postgres') {
+                    agent { label 'fast' }
+                    environment {
+                        RUBY = '2.5'
+                        GEMSET = 'ruby-2.5-postgres'
+                    }
+                    stages {
+                        stage("setup-2.5-postgres") {
+                            steps {
+                                git url: 'https://github.com/theforeman/foreman', branch: 'develop'
```

Comment:
We do repeat the branch and URL quite a few times. How about variables? Since we copy the pipeline on branching it'd make that part easier.

Reply:
Agreed on branch since that can be configured. The URL I am reticent since that shouldn't really change and this is just as or more readable than a variable.